parmed/amber/_amberparm.py

```diff
@@ -387,17 +387,17 @@ def from_structure(cls, struct, copy=False):
         # present as a way to hack entries into the 1-4 pairlist. See
         # https://github.com/ParmEd/ParmEd/pull/145 for discussion. The solution
         # here is to simply set that periodicity to 1.
-        for dt in inst.dihedral_types:
-            if dt.phi_k == 0 and dt.per == 0:
-                dt.per = 1.0
-            elif dt.per == 0:
-                warn('Periodicity of 0 detected with non-zero force constant. '
-                     'Changing periodicity to 1 and force constant to 0 to '
-                     'ensure 1-4 nonbonded pairs are properly identified. This '
-                     'might cause a shift in the energy, but will leave forces '
-                     'unaffected', AmberWarning)
-                dt.phi_k = 0.0
-                dt.per = 1.0
+#        for dt in inst.dihedral_types:
+#            if dt.phi_k == 0 and dt.per == 0:
+#                dt.per = 1.0
+#            elif dt.per == 0:
+#                warn('Periodicity of 0 detected with non-zero force constant. '
+#                     'Changing periodicity to 1 and force constant to 0 to '
+#                     'ensure 1-4 nonbonded pairs are properly identified. This '
+#                     'might cause a shift in the energy, but will leave forces '
+#                     'unaffected', AmberWarning)
+#                dt.phi_k = 0.0
+#                dt.per = 1.0
```

Comment:
It *is* a hack, but it's a hack that is needed to work around an implicit assumption about the Amber prmtop file encoded within pmemd.

The consequence of eliminating this code is that 1-4 interactions may be computed incorrectly (which is why this code was added in the first place).  We need to find some way to avoid introducing this known bug.  Another option would be to print the constant energy shift as part of the warning so it can be validated by hand.

Note that a constant energy shift makes no difference in any thermodynamical (or dynamical) property, so I'm more comfortable having energies differ and forces remain constant sometimes than getting them wrong (subtly and silently) in pmemd (and pmemd.cuda) later...